_plugins/bidirectional_links_generator.rb

```diff
@@ -39,108 +39,50 @@ def generate(site)
-      notes_linking_to_current_note.each do |n|
-        graph_edges << {
-          source: note_id_from_note(n),
-          target: note_id_from_note(current_note),
-        }
-
-        if current_note.data['tags'].include?("CrossFit")
-          crossfit_graph_nodes << {
+      sub_graph_nodes.each {|key, value| 
+        if current_note.data['tags'].include?("#{key}")
+          value << {
             id: note_id_from_note(current_note),
             path: current_note.url,
             label: current_note.data['title'],
           } unless current_note.path.include?('_notes/index.html')
         end
+        value.uniq!
+      }
 
-        if current_note.data['tags'].include?("CrossFit")
-          notes_linking_to_current_note.each do |n|
-            if n.data['tags'].include?("CrossFit")
-            crossfit_graph_edges << {
-              source: note_id_from_note(n),
-              target: note_id_from_note(current_note),
-            }
-            end
-          end
-        end
-
-        if current_note.data['tags'].include?("design")
-          design_graph_nodes << {
-            id: note_id_from_note(current_note),
-            path: current_note.url,
-            label: current_note.data['title'],
-          } unless current_note.path.include?('_notes/index.html')
-        end
-
-        if current_note.data['tags'].include?("design")
-          notes_linking_to_current_note.each do |n|
-            if n.data['tags'].include?("design")
-              design_graph_edges << {
-              source: note_id_from_note(n),
-              target: note_id_from_note(current_note),
-            }
-            end
-          end
-        end
-
-
-        if current_note.data['tags'].include?("nutrition")
-          nutrition_graph_nodes << {
-            id: note_id_from_note(current_note),
-            path: current_note.url,
-            label: current_note.data['title'],
-          } unless current_note.path.include?('_notes/index.html')
-        end
+			# Edges: Jekyll
+      current_note.data['backlinks'] = notes_linking_to_current_note
 
-        if current_note.data['tags'].include?("nutrition")
-          notes_linking_to_current_note.each do |n|
-            if n.data['tags'].include?("nutrition")
-              nutrition_graph_edges << {
-              source: note_id_from_note(n),
-              target: note_id_from_note(current_note),
-            }
+      # Edges: Graph
+        notes_linking_to_current_note.each do |n|
+          sub_graph_edges.each {|key, value|
+            if current_note.data['tags'].include?("#{key}")
+              notes_linking_to_current_note.each do |n|
+                if n.data['tags'].include?("#{key}")
+                  value << {
+                    source: note_id_from_note(n),
+                    target: note_id_from_note(current_note),
+                  }
+                end
+                value.uniq!
+              end
             end
-          end
+          }
```

Comment:
Same logic for edges.

Have to do two checks because you need to see if the current node is tagged as well as the linked node is tagged.

Linked nodes have to be tagged with the same tag because currently the nodes array won't have nodes that are not tagged.

Alternatively, I could check to see if a current note is tagged, then if the linked note is not tagged, add the node then add the edge.